### PR TITLE
#167729790 Ft logout user session 

### PR DESCRIPTION
**Pull request by davidfrank96**
#### What does this PR do?
#This is a logout endpoint were users can logout
#### Description of Task to be completed?
#- created a logout.,js file in the controller
#-updated the index file for routes
#-created a logout.route.js file 
#### How should this be manually tested?
N/A
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#167729790
#### Questions:
N/A

### CODE REVIEW

File: src/database/migrations/20190818182539-create-user.js [hunk 1]
```diff
@@ -32,5 +32,5 @@ export default {
       type: Sequelize.DATE
     }
   }),
-  down: queryInterface => queryInterface.dropTable('Users')
+  down: (queryInterface) => queryInterface.dropTable('Users')
```
[the remark this answers is not in the document]
Comment (chibytez) [reply src/database/migrations/20190818182539-create-user.js]: Nice work, can we remove the parentheses around queryInterface

File: src/database/seeders/20190817154557-User.js [hunk 2]
```diff
@@ -25,5 +25,5 @@ export default {
     {},
 
   ),
-  down: queryInterface => queryInterface.bulkDelete('Users', null, {}),
+  down: (queryInterface) => queryInterface.bulkDelete('Users', null, {}),
```
[the remark this answers is not in the document]
Comment (chibytez) [reply src/database/seeders/20190817154557-User.js]: can we remove the parentheses around queryInterface

File: src/database/seeders/20190817154557-User.js [hunk 3]
```diff
@@ -2,7 +2,7 @@
 import bcrypt from 'bcryptjs';
 
 export default {
-  up: queryInterface => queryInterface.bulkInsert(
+  up: (queryInterface) => queryInterface.bulkInsert(
```
[the remark this answers is not in the document]
Comment (chibytez) [reply src/database/seeders/20190817154557-User.js]: also, remove the parentheses around queryInterface

File: src/database/migrations/20190818182539-create-user.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidfrank96) [reply src/database/migrations/20190818182539-create-user.js]: okay...

File: src/controllers/logout.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+
+/**
+ *@description A controller that destroys sessions and redirects users to the homepage
+ *
+ * @class Logout
+ */
+
+class Logout {
```
[the remark this answers is not in the document]
Comment (chibytez) [reply src/controllers/logout.js:7]: can we reduce the line space between your JSDoc and your class, and make the jsDoc look like this
/**
 * @description class description
 * @class className
 */

File: src/controllers/logout.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  /**
```
[the remark this answers is not in the document]
Comment (okabamac) [reply src/controllers/logout.js:8]: Try to follow the jsDoc Chibuike provided below. That will help

File: src/controllers/logout.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+
+/**
+ * @class Logout
+ * @description Performs logout related related tasks
+ * @exports Logout
+ */
+class Logout {
```
[the remark this answers is not in the document]
Comment (okabamac) [reply src/controllers/logout.js:7]: Did you use this class?

[the remark this answers is not in the document]
Comment (davidfrank96) [reply src/controllers/logout.js:7]: please can you elaborate on that pls

**Review by okafor-chidimma — APPROVED** [review]
Just pulled your branch and tested...all test cases passed...I approve for merge

**Review by okafor-chidimma — CHANGES_REQUESTED** [review]
edit your eslint.json not to allow arrow-parens

**Review by okafor-chidimma — APPROVED** [review]
Looks good and I have tested it locally. Passed all test

**Review by chibytez — APPROVED** [review]
looks good

File: src/index.js [hunk 7]
```diff
@@ -31,22 +32,25 @@ app.use(methodOverride());
...
+}));
 
 if (!isProduction) {
   app.use(errorhandler());
 }
 
-
 app.use(apiRoutes);
 
+app.use(session({
```
Comment (Stylll) [inline src/index.js]: Why do we have this repeated here?

File: src/routes/api/index.js [hunk 8]
```diff
@@ -1,31 +1,42 @@
...
+
+//     if (user) {
+//       return res.json({ user: user.toAuthJSON() });
+//     }
+//     return res.status(422).json(info);
+//   })(req, res, next);
+// });
+
+
+// export default usersRouter;
```
Comment (Stylll) [inline src/routes/api/index.js]: I see you commented these lines out. If they are not useful, you should take them out?. I noticed @chibytez and @okafor-chidimma also approved this PR. Do you think the commented lines should be left in the codebase?

File: src/test/app.test.js [hunk 9]
```diff
@@ -16,3 +16,27 @@ describe('TESTING ENDPOINTS', () => {
...
+      .get('/api/v1/auth/logout')
+      .end((err, res) => {
+        if (err) return done(err);
+        expect(res.statusCode).to.equal(200);
+        expect((res.text)).to.be.a('string');
+        done();
+      });
+  });
+
+  it('should return "you have logged out successfully" Page', (done) => {
```
Comment (Stylll) [inline src/test/app.test.js]: You have duplicate test cases. Please remove one.

**Review by Stylll — CHANGES_REQUESTED** [review]
@davidfrank96 Left a few comments. Please check them out. cc: @chibytez @okafor-chidimma

File: src/routes/api/index.js (hunk 8 above)
Comment (davidfrank96) [reply src/routes/api/index.js]: it will be taken off... I think this code is redundant

File: src/test/app.test.js [hunk 10]
```diff
@@ -16,3 +16,16 @@ describe('TESTING ENDPOINTS', () => {
       });
   });
 });
+
+describe('TESTING LOGOUT', () => {
```
Comment (Stylll) [inline src/test/app.test.js]: we should move this test suite to its own file

**Review by Stylll — CHANGES_REQUESTED** [review]
@davidfrank96 I dropped a comment. Kindly review and also fix the merge conflicts.

File: src/test/app.test.js (hunk 10 above)
Comment (davidfrank96) [reply src/test/app.test.js]: okay will do now...

File: src/routes/api/logout.route.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import { Router } from 'express';
+
+import Logout from '../../controllers/logout';
+
+const router = Router();
+const { logout } = Logout;
+
+router.get('/v1/auth/logout', logout);
```
Comment (Stylll) [inline src/routes/api/logout.route.js]: I just noticed the logout route is a `GET` request. Is there a reason why this is not a POST request?.
Also, I see you named your route file `logout.route.js`, every route file in the codebase doesn't have the `**.route.js`. Please update your code to follow the existing pattern.

**Review by Stylll — CHANGES_REQUESTED** [review]
@davidfrank96 new changes request. Please review.

File: src/routes/api/logout.route.js (hunk 11 above)
Comment (davidfrank96) [reply src/routes/api/logout.route.js]: on the passport docs it was a get request that's why i followed suit

File: src/index.js [hunk 12]
```diff
@@ -7,6 +7,7 @@ import cors from 'cors';
 import errorhandler from 'errorhandler';
 import morgan from 'morgan';
 import methodOverride from 'method-override';
+
```
Comment (Stylll) [inline src/index.js]: remove empty space

File: src/index.js [hunk 13]
```diff
@@ -34,6 +35,7 @@ app.use(express.static(`${__dirname}/public`));
 app.use(passport.initialize());
 app.use(passport.session());
 
+
```
Comment (Stylll) [inline src/index.js:37]: remove empty space

File: src/routes/api/index.js [hunk 14]
```diff
@@ -1,11 +1,15 @@
-
+// import { Router } from 'express';
```
Comment (Stylll) [inline src/routes/api/index.js]: remove the commented line

File: src/routes/api/index.js [hunk 15]
```diff
@@ -1,11 +1,15 @@
...
 import express from 'express';
 import socialMediaRoutes from './socialmedia';
 import authRouter from './auth';
+import logoutRoute from './logout';
 
 const usersRouter = express.Router();
 usersRouter.use('/', socialMediaRoutes);
 
+// const usersRouter = Router();
+
```
Comment (Stylll) [inline src/routes/api/index.js]: remove the commented line and empty space

**Review by Stylll — CHANGES_REQUESTED** [review]
Please pay attention and ensure to review your code properly. There are a few empty spaces and commented lines. I dropped comments on them, kindly remove them.

**Review by Stylll — APPROVED** [review]

**Pull request merged**